lib/SILGen/SILGenPattern.cpp

```diff
@@ -2151,37 +2146,20 @@ void PatternMatchEmission::emitEnumElementDispatch(
 
       // If the payload is boxed, project it.
       if (elt->isIndirect() || elt->getParentEnum()->isIndirect()) {
```

Comment:
Notice how here we check for getParentEnum().isIndirect. That is the case to check.

Reply:
@gottesmm that is the test case that I added.